pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -186,11 +189,40 @@ func dropTopologyOnV1(ctx context.Context, oldEPS, newEPS *discovery.EndpointSli
 			return
 		}
 
+		oldDeprecatedTopologyNodeNames := getDeprecatedTopologyNodeNames(oldEPS)
+
 		for i := range newEPS.Endpoints {
 			ep := &newEPS.Endpoints[i]
 
-			//Silently clear out DeprecatedTopology
-			ep.DeprecatedTopology = nil
+			if ep.DeprecatedTopology != nil {
```

Comment:
no need for a nil check, lookup on nil map returns `zero-value, false`

Reply:
Updated, thanks!